pkg/inspector/image-inspector.go

```diff
@@ -210,27 +223,33 @@ func (i *defaultImageInspector) pullImage(client *docker.Client) error {
 		return authCfgErr
 	}
 
-	reader, writer := io.Pipe()
-	// handle closing the reader/writer in the method that creates them
-	defer writer.Close()
-	defer reader.Close()
-	imagePullOption := docker.PullImageOptions{
-		Repository:    i.opts.Image,
-		OutputStream:  writer,
-		RawJSONStream: true,
-	}
-
-	bytesChan := make(chan int)
-	go aggregateBytesAndReport(bytesChan)
-	go decodeDockerPullMessages(bytesChan, reader)
-
 	// Try all the possible auth's from the config file
 	var authErr error
 	for name, auth := range imagePullAuths.Configs {
-		if authErr = client.PullImage(imagePullOption, auth); authErr == nil {
+		reader, writer := io.Pipe()
+		// handle closing the reader/writer in the method that creates them
+		defer writer.Close()
+		defer reader.Close()
+		imagePullOption := docker.PullImageOptions{
+			Repository:    i.opts.Image,
+			OutputStream:  writer,
+			RawJSONStream: true,
+		}
+
+		bytesChan := make(chan int, 100)
+		parsedErrors := make(chan bool, 1)
+		go decodeDockerPullMessages(parsedErrors, bytesChan, reader)
+
+		if authErr := client.PullImage(imagePullOption, auth); authErr != nil {
+			return authErr
+		}
+		writer.Close()
+		reader.Close()
+		if parsedError := <-parsedErrors; parsedError {
```

Comment:
the way that `decodeDockerPullMessages` is coded seems to indicate that decoding can happen many times and that multiple calls to the `parsedErrors` channel may happen.  I think that more strongly indicates that either we need to aggregate errors in `decodeDockerPullMessages` or this would need to change.  

It has been a while since I've been in this code so maybe this is a bad question: do we know why the `v.Error` doesn't surface in the `client.PullImage` call since it returns an error?  It seems weird that we'd have to have both the error check and parsing the progress stream for errors.